### ISSUE TITLE
[Documentation] Ease distribution of games made with custom builds

### ISSUE THREAD

**Issue by scambier** [nesbox/TIC-80#2168]
As of today, using the `export` command to distribute a game a) downloads a binary from tic80.com, and b) uses custom binaries for each scripting language (assuming `alone=1`). So if you'd like to distribute a game with a custom tic80 build, you need to
1. Make a distributable build from your fork, but the process isn't documented
2. Have a http server to host this build so that TIC-80 can download it

I suppose the 2nd point is easily solvable with a localhost server (though a bit cumbersome), but making distributable builds looks quite obscure without the documentation, for someone who is not familiar with the project's build process (or even C projects in general).

Would it be possible to get some basic documentation on how to make these builds? Thank you :)

**Comment by scambier** [comment]
So I tried a few things. To make the "alone" builds, I added `-DBUILD_STUB=ON -DBUILD_PLAYER=ON`. I'm not sure if both vars are required, but it definitely builds 1 executable per language.

Surprisingly, I'm having more issues to change the source of downloaded builds.
- I changed `TIC_HOST` to my own domain, and uploaded the files with the same folder structure
- Full rebuild of TIC-80
- `export win win alone=1` 
- The nginx log looks correct: `[19/Apr/2023:12:11:38 +0200] "GET /export/1.1-dev/winlua HTTP/1.1" 302 161 "-" "-"`
- ![image](https://user-images.githubusercontent.com/3216752/233043775-ec416ba2-bca8-4e20-afba-974359d151cd.png)

The error is triggered from console.c inside `onExportGet`. I'm not sure what's going on and it's quite hard to follow the trail with my lack of knowledge in C.

**Comment by scambier** [comment]
So here's what I found so far. I guess it's good enough for a wiki page, but I'd like to solve the last missing piece: how to bypass the need of a web server. It's probably a simple command to append your game to the executable.

https://gist.github.com/scambier/a72f8dafa1f7b9fd3ca136125a54c21b

**Comment by MineRobber9000** [comment]
Hi! I've done custom standalone builds before, and it's not really that complicated? It's still complicated, mind you, but it's somewhat straightforward (copied from Discord):

> * keep the binary's size handy
> * zip the cart (tic_tool_zip uses compress2 with Z_BEST_COMPRESSION)
> * append "TIC.CART", the length of the binary (pre-appending), the length of the cart (zipped), and then the zipped cart
> 
> the specific check on the sizes is that `appSize == header->appSize + sizeof(EmbedHeader) + header->cartSize`
